libs/utils/src/lib.rs

```diff
@@ -95,3 +95,11 @@ macro_rules! project_git_version {
         );
     };
 }
+
+/// Same as `assert!`, but evaluated during compilation and gets optimized out in runtime.
+#[macro_export]
+macro_rules! const_assert {
+    ($($args:tt)*) => {
+        const _: () = assert!($($args)*);
+    };
+}
```

Comment:
There is a popular `static_assertions` crate, but it [wasn't updated for a while](https://github.com/nvzqz/static-assertions-rs/issues/43).